overlord/configstate/configcore/tmpfs_test.go

```diff
@@ -0,0 +1,197 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/overlord/configstate/configcore"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type tmpfsSuite struct {
+	configcoreSuite
+
+	servOverridePath string
+	servOverrideDir  string
+}
+
+var _ = Suite(&tmpfsSuite{})
+
+func (s *tmpfsSuite) SetUpTest(c *C) {
+	s.configcoreSuite.SetUpTest(c)
+
+	s.servOverrideDir = filepath.Join(dirs.SnapServicesDir, "tmp.mount.d")
+	s.servOverridePath = filepath.Join(s.servOverrideDir, "override.conf")
+}
+
+// Configure with different valid values
+func (s *tmpfsSuite) TestConfigureTmpfsGoodVals(c *C) {
+	mountCmd := testutil.MockCommand(c, "mount", "")
+	mountCalls := [][]string{}
+
+	for _, size := range []string{"100m", "1g", "512k", "104857600",
+		"2M", "7G", "1024K", "20%"} {
+
+		err := configcore.Run(coreDev, &mockConf{
+			state: s.state,
+			conf: map[string]interface{}{
+				"tmpfs.size": size,
+			},
+		})
+		c.Assert(err, IsNil)
+
+		c.Check(s.servOverridePath, testutil.FileEquals,
+			fmt.Sprintf("[Mount]\nOptions=mode=1777,strictatime,nosuid,nodev,size=%s\n",
+				size))
+		mntOpts := fmt.Sprintf("remount,mode=1777,strictatime,nosuid,nodev,size=%s",
+			size)
+		mountCalls = append(mountCalls, []string{"mount", "-o", mntOpts, "/tmp"})
+	}
+
+	c.Check(s.systemctlArgs, DeepEquals, [][]string(nil))
```

Comment:
(nitpick) `c.Check(s.systemctlArgs, HasLen, 0)` would also work and might be slightly easier but either way is fine really.